x-pack/legacy/plugins/apm/public/utils/formatters.ts

```diff
@@ -245,3 +249,145 @@ export const getFixedByteFormatter = memoize((max: number) => {
 
   return bailIfNumberInvalid(formatter);
 });
+
+/**
+ * Returns the timezone set on momentTime.
+ * (UTC+offset) when offset if bigger than 0.
+ * (UTC-offset) when offset if lower than 0.
+ * @param momentTime Moment
+ */
+function formatTimezone(momentTime: moment.Moment) {
+  const DEFAULT_TIMEZONE_FORMAT = 'Z';
+
+  // Adds plus sign when offsetHours is greater than 0.
+  const getCustomTimezoneFormat = (offsetHours: number) =>
+    offsetHours > 0 ? `+${offsetHours}` : offsetHours;
+
+  const utcOffsetHours = momentTime.utcOffset() / 60;
+
+  const utcOffsetFormatted = Number.isInteger(utcOffsetHours)
+    ? getCustomTimezoneFormat(utcOffsetHours)
+    : DEFAULT_TIMEZONE_FORMAT;
+
+  return momentTime.format(`(UTC${utcOffsetFormatted})`);
+}
+
+export type TimePrecision = 'hours' | 'minutes' | 'seconds' | 'milliseconds';
+function getTimeFormat(precision: TimePrecision) {
+  switch (precision) {
+    case 'hours':
+      return 'HH';
+    case 'minutes':
+      return 'HH:mm';
+    case 'seconds':
+      return 'HH:mm:ss';
+    case 'milliseconds':
+      return 'HH:mm:ss.SSS';
+    default:
+      return '';
+  }
+}
+
+type DatePrecision = 'days' | 'months' | 'years';
+function getDateFormat(precision: DatePrecision) {
+  switch (precision) {
+    case 'years':
+      return 'YYYY';
+    case 'months':
+      return 'MMM YYYY';
+    case 'days':
+      return 'MMM D, YYYY';
+    default:
+      return '';
+  }
+}
+
+function getFormatsAccordingToDateDifference(
```

Comment:
Maybe simpler is better for now

```suggestion
function getDateTimeFormats(
```

If we get another format returning function that doesn't take date difference into account we can make it clear in the name. But for now I don't think it's necessary.

Reply:
That was the name actually, I decided to change it because we already have 'getDateFormat' and I didn't want to associate them.